source/MaterialXView/Material.cpp

```diff
@@ -528,11 +528,34 @@ void Material::drawPartition(mx::MeshPartitionPtr part) const
 
 mx::VariableBlock* Material::getPublicUniforms() const
 {
-    return _hwShader ? mx::getUniformBlock(_hwShader, mx::Stage::PIXEL, mx::HW::PUBLIC_UNIFORMS) : nullptr;
+    if (!_hwShader)
+    {
+        return nullptr;
+    }
+    try
+    {
+        mx::ShaderStage& stage = _hwShader->getStage(mx::Stage::PIXEL);
+        mx::VariableBlock& block = stage.getUniformBlock(mx::HW::PUBLIC_UNIFORMS);
+        return &block;
+    }
+    catch (mx::Exception& /*e*/)
+    {
+        // Pass-though. Caller weill handle null return values.
```

Comment:
I still think catching this exception is wrong. If there is no pixel stage or no uniform block it's a fatal error we shouldn't hide.